openwisp_users/tests/test_admin.py

```diff
@@ -1275,6 +1275,76 @@ def test_only_superuser_can_delete_inline_org_owner(self):
+
+        with self.subTest('test delete org user which belongs to owner'):
+            post_data = {'post': 'yes'}
+            url = reverse(
+                f'admin:{self.app_label}_organizationuser_delete', args=[org_user.pk]
+            )
+            r = self.client.post(url, post_data, follow=True)
+            qs = OrganizationUser.objects.filter(organization=org1, user=user1)
+            self.assertEqual(r.status_code, 200)
+            msg = (
+                't delete this organization user because it '
+                'belongs to an organization owner'
+            )
+            self.assertContains(r, msg)
+            self.assertEqual(qs.count(), 1)
+
+        with self.subTest('test delete org user which belongs to no owner'):
+            org2 = self._create_org(name='org2')
+            org_u = self._create_org_user(user=user1, organization=org2, is_admin=False)
+            post_data = {'post': 'yes'}
+            url = reverse(
+                f'admin:{self.app_label}_organizationuser_delete', args=[org_u.pk]
+            )
+            r = self.client.post(url, post_data, follow=True)
+            qs = OrganizationUser.objects.filter(organization=org2, user=user1)
+            self.assertEqual(r.status_code, 200)
+            self.assertContains(r, 'was deleted successfully.')
+            self.assertEqual(qs.count(), 0)
+
+        with self.subTest("Can not delete only owner's org user with action"):
+            post_data = {
+                'action': 'delete_selected_overridden',
+                '_selected_action': [org_user.pk],
+                'post': 'yes',
+            }
+            url = reverse(f'admin:{self.app_label}_organizationuser_changelist')
+            r = self.client.post(url, post_data, follow=True)
+            qs = OrganizationUser.objects.filter(user=user1, organization=org1)
+            self.assertEqual(r.status_code, 200)
+            self.assertContains(r, 't delete organization users which belong to owners')
+            self.assertEqual(qs.count(), 1)
+
+        with self.subTest('delete org users with some belonging to owners'):
+            org2 = self._create_org(name='org2')
+            org_user2 = self._create_org_user(user=user1, organization=org2)
+            post_data = {
+                'action': 'delete_selected_overridden',
+                '_selected_action': [org_user.pk, org_user2.pk],
+            }
+            url = reverse(f'admin:{self.app_label}_organizationuser_changelist')
+            r = self.client.post(url, post_data, follow=True)
+            self.assertEqual(r.status_code, 200)
+            msg = (
+                't delete 1 organization user because '
+                'it belongs to an organization owner'
+            )
+            self.assertContains(r, msg)
+            post_data.update({'post': 'yes'})
+            r = self.client.post(url, post_data, follow=True)
```

Comment:
This is the point which calls `delete_selected_overridden`. with the `delete_selected_overridden`, the number of queries this line shows is `17` and when `delete_selected_overridden` is commented out (that is using the default delete_selected) the number of queries returned is `16`. and the extra query can be accounted for by the line which does `queryset = queryset.exlcude(pk_in=pks)`

Reply:
cool, please add the `assertNumQueries` both here and above.
If there's any discrepancies due to the django version used, you can follow this approach: https://github.com/openwisp/openwisp-network-topology/blob/dff2357602d24f24b71e3e7aa83d61e8b94718af/openwisp_network_topology/integrations/device/tests.py#L93-L96